reportssql/treatment_report_functions.sql

```diff
@@ -431,3 +434,37 @@ BEGIN
     RETURN (result );
 END$$ 
 DELIMITER ;
+
+-- patientARVDrugDurationBetween
+
+DROP FUNCTION IF EXISTS patientARVDrugDurationBetween;
+
+DELIMITER $$
+CREATE FUNCTION patientARVDrugDurationBetween(
+    p_patientId INT(11),
+    p_drugDurationMonthsMin INT(11),
+    p_drugDurationMonthsMax INT(11)) RETURNS TINYINT(1)
+    DETERMINISTIC
+BEGIN
+
+DECLARE result TINYINT(1) DEFAULT 0;
+
+    SELECT true INTO result
+    FROM orders o
+    JOIN drug_order do ON do.order_id = o.order_id
+    JOIN concept c ON do.duration_units = c.concept_id AND c.retired = 0
+    JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+    WHERE o.patient_id = p_patientId AND o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND calculateTreatmentEndDate(
+            o.scheduled_date,
+            do.duration,
+            c.uuid -- uuid of the duration unit concept
+            ) BETWEEN timestampadd(MONTH, p_drugDurationMonthsMin, o.scheduled_date) 
+                AND timestampadd(MONTH, p_drugDurationMonthsMax, o.scheduled_date)
```

Comment:
@lauravignoli the current implementation will not work well when the duration is a decimal.
If for example, we need to check that the duration is between 0 and 2 months; and there is a drug ordered on 10 Jan with a duration of 2 months and a half; the math will be:
- treatmentEndDate = 25 Mar (10 Jan + 2 months and 15 days)
- interval1 = 10 Jan (10 Jan + 0 month)
- interval2 = 10 Mar (10 Jan + 2 months)

25 Mar  BETWEEN 10 Jan AND 10 Mar => False

A possible fix is to create a function that calculates 'do.duration' to months (that discards the decimal, i.e. 2.8 months should return 2 months); and the condition could be something like:
``` getDurationInMonths(do.duration, c.uuid) BETWEEN p_drugDurationMonthsMin AND p_drugDurationMonthsMax```

Reply:
@tmvumbi2 Why would we use the duration with decimals? If you want we can add a comment that the months must be integer values?

Reply:
@lauravignoli sorry my comment wasn't clear enough, it was referring to the duration of the  treatment (i.e. not the interval provided as parameters)